### PR TITLE
Added friendlier dsql error message for 405 (which occurs when druid.sql.enabled=false)

### PR DESCRIPTION
**Pull request by justinborromeo**
Small quality of life fix that prompts user to check if `druid.sql.enabled` is `true` on brokers if a 405 is returned.  

One question: is there any other case where the sql endpoint would return a 405 for a non-config related reason?

### CODE REVIEW

**Comment by gianm** [comment]
Actually, now that I think about it, maybe change the message to encourage people to both check `druid.sql.enabled` and check that the URL is correct.

**Comment by justinborromeo** [comment]
Done.  Error message is: `HTTP Error 405: Method Not Allowed - Are you using the correct broker URL and is druid.sql.enabled set to true on your broker?`

**Comment by justinborromeo** [comment]
CI's good

File: examples/bin/dsql-main [hunk 1]
```diff
@@ -136,6 +136,10 @@ def raise_friendly_error(e):
       if error_obj['host']:
         error_text = error_text + ' (' + str(error_obj['host']) + ')'
       raise DruidSqlException(error_text)
+    elif e.code == 405:
+      error_text = 'HTTP Error {0}: {1}\n{2}'.format(e.code, e.reason + " - Are you using the correct broker URL and " +\
```
Comment (asdf2014) [inline examples/bin/dsql-main]: The backslash is redundant here.

**Pull request merged**